src/hotspot/share/gc/g1/g1HeapRegion.cpp

```diff
@@ -335,6 +338,24 @@ class VerifyCodeRootOopClosure: public OopClosure {
   bool has_oops_in_region() { return _has_oops_in_region; }
 };
 
+
+double G1HeapRegion::total_based_on_incoming_refs_ms() {
+
+  G1Policy* p = G1CollectedHeap::heap()->policy();
+
+  double merge_scan_time_ms = p->predict_merge_scan_time(_incoming_refs); // We use the number of incoming references as an estimate for remset cards.
+  double non_young_other_time_ms = p->predict_non_young_other_time_ms(1);
+  double predicted_copy_time_ms = p->predict_region_copy_time_ms(this, false /* for_young_only_phase */);
+
+  return merge_scan_time_ms +
+         non_young_other_time_ms +
+         predicted_copy_time_ms;
+}
+
+double G1HeapRegion::gc_efficiency() {
+  return reclaimable_bytes() / total_based_on_incoming_refs_ms();
+}
```

Comment:
I wonder if `total_based_on_incoming_refs_ms` can be inlined to its single caller. Also, these logic doesn't belong to a heap-region -- maybe `G1Policy`, since most of code use `p->`?